velox/functions/sparksql/tests/SparkPartitionIdTest.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/sparksql/tests/SparkFunctionBaseTest.h"
+
+namespace facebook::velox::functions::sparksql::test {
+namespace {
+
+class SparkPartitionIdTest : public SparkFunctionBaseTest {
+ protected:
+  void testSparkPartitionId(int32_t partitionId, int32_t vectorSize) {
+    queryCtx_->testingOverrideConfigUnsafe(
+        {{core::QueryConfig::kSparkPartitionId, std::to_string(partitionId)}});
+    auto result =
+        evaluate("spark_partition_id()", makeRowVector(ROW({}), vectorSize));
+    velox::test::assertEqualVectors(
+        makeFlatVector<int32_t>(
+            vectorSize, [&](auto row) { return partitionId; }),
+        result);
+  }
+
+  void testSparkPartitionIdFails(const std::string& errorMsg) {
```

Comment:
Remove this function and use VELOX_ASSERT_THROW directly. This will produce better error messages pointing clearly to the test case that failed.

Reply:
Updated, thanks.